docs/integration-guide/request-interceptors.md

```diff
@@ -0,0 +1,417 @@
+---
+title: Request Interceptors
+---
+
+Request Interceptors provide a powerful way to customize request and response handling in the Gusto Embedded React SDK. The SDK provides four distinct hook types to handle different phases of the HTTP request lifecycle. This allows you to implement authentication, logging, error handling, and other cross-cutting concerns with precision and flexibility.
```

Comment:
Can we link out to the speakeasy reference here? it might get stale less quickly that way